src/panels/lovelace/entity-rows/hui-media-player-entity-row.ts

```diff
@@ -0,0 +1,197 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-icon-button/paper-icon-button";
+
+import "../components/hui-generic-entity-row";
+
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { EntityRow, EntityConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+import { HassEntity } from "home-assistant-js-websocket";
+
+const SUPPORT_PAUSE = 1;
+const SUPPORT_NEXT_TRACK = 32;
+const SUPPORTS_PLAY = 16384;
+const OFF_STATES = ["off", "idle"];
+
+class HuiMediaPlayerEntityRow extends hassLocalizeLitMixin(LitElement)
+  implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: EntityConfig;
+
+  static get properties() {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-error-entity-row
+          .entity="${this._config.entity}"
+        ></hui-error-entity-row>
+      `;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <hui-generic-entity-row
+        .hass="${this.hass}"
+        .config="${this._config}"
+        .showSecondary="false"
+      >
+        ${
+          !this._isOff(stateObj.state)
+            ? html`
+                <div class="controls">
+                  ${
+                    this._computeControlIcon(stateObj)
+                      ? html`
+                          <paper-icon-button
+                            icon="${this._computeControlIcon(stateObj)}"
+                            @click="${this._playPause}"
+                          ></paper-icon-button>
+                        `
+                      : ""
+                  }
+                  ${
+                    this._supportsNext(stateObj)
+                      ? html`
+                          <paper-icon-button
+                            icon="hass:skip-next"
+                            @click="${this._nextTrack}"
+                          ></paper-icon-button>
+                        `
+                      : ""
+                  }
+                </div>
+              `
+            : ""
+        }
+        ${
+          this._isOff(stateObj.state)
```

Comment:
This is confusing and also doing the check again. I would suggest you move all this logic above the return statement and also remove all the Polymer helper functions. We don't need them and they just spread our logic over 1000 small one-line functions instead of keeping it together:

```ts
let content;
const state = stateObj.state;

if (OFF_STATES.includes(state)) {
    content = html`<div>${(
      this.localize(`state.media_player.${state}`) ||
      this.localize(`state.default.${state}`) ||
      state
    )}</div>`;
} else if (state !== "playing") {
  content = …
}

return html`<hui-generic-entity-row>${content}</hui-generic-entity-row>`
```